astropy/utils/compat/optional_deps.py

```diff
@@ -0,0 +1,37 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Checks for optional dependencies using lazy import from
+`PEP 562 <https://www.python.org/dev/peps/pep-0562/>`_.
+"""
+import importlib
+
+# First, the top-level packages:
+# TODO: This list is a duplicate of the dependencies in setup.cfg "all", but
+# some of the package names are different from the pip-install name (e.g.,
+# beautifulsoup4 -> bs4).
+_optional_deps = ['bleach', 'bottleneck', 'bs4', 'bz2', 'h5py', 'html5lib',
+                  'IPython', 'jplephem', 'lxml', 'matplotlib', 'mpmath',
+                  'pandas', 'PIL', 'pkg_resources', 'pytz', 'scipy', 'skyfield',
+                  'sortedcontainers', 'lzma', 'yaml']
+_deps = {k: k for k in _optional_deps}
```

Comment:
I would (somewhat strongly) suggest using the upper-case version with `_deps = {k.upper(): k for k in _optional_deps}` here. Rationale:
- Substantially reduces change footprint. Don't change code that isn't broken and has 10 years of precedent.
- We've always used these as module-level constants, which are SCREAMING_SNAKE_CASE per PEP8. Seeing `HAS_lxml` falls in no-person's land. 
- The only place we could get in trouble is if we have two optional deps that are named differently only by case. I strongly doubt this will ever happen but if does it could be special-cased like `plt` below.

Reply:
I guess that makes sense! I wanted to avoid special-casing PIL and IPython, but only realized after submitting this that  the dictionary makes it very easy as you show, so I will do the replace. Before I finalize I'll rebase to get rid of the change/change back but for now I'll just push up the replacements.

Reply:
I did the find/replace to upper case and ended up doing the rebase to simplify the commits. Unfortunately, now most of the work is done in one big commit. But scrolling through files changed, it doesn't look too bad to review 😅